packages/arb-tx-aggregator/txaggregator/server.go

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2020, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package txaggregator
+
+import (
+	"context"
+	"errors"
+	"log"
+	"math/big"
+	"sync"
+	"time"
+
+	errors2 "github.com/pkg/errors"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/offchainlabs/arbitrum/packages/arb-util/common"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/hashing"
+	"github.com/offchainlabs/arbitrum/packages/arb-validator-core/arbbridge"
+	"github.com/offchainlabs/arbitrum/packages/arb-validator-core/message"
+)
+
+const maxTransactions = 50
+
+const signatureLength = 65
+const recoverBitPos = signatureLength - 1
+
+type Server struct {
+	rollupAddress common.Address
+	globalInbox   arbbridge.GlobalInbox
+
+	sync.Mutex
+	valid        bool
+	transactions []message.Transaction
+	signatures   [][signatureLength]byte
+}
+
+// NewServer returns a new instance of the Server class
+func NewServer(
+	ctx context.Context,
+	globalInbox arbbridge.GlobalInbox,
+	rollupAddress common.Address,
+) *Server {
+	server := &Server{
+		rollupAddress: rollupAddress,
+		globalInbox:   globalInbox,
+		valid:         true,
+	}
+
+	go func() {
+		ticker := time.NewTicker(common.NewTimeBlocksInt(5).Duration())
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [avoid magic numbers like '5', create a named constant for it](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5428340)